build/Jenkinsfile.pr

```diff
@@ -0,0 +1,264 @@
+
+                                curl -f -o ./dist.tar.gz https://releases.mattermost.com/mattermost-webapp/commit/\${WEBAPP_GIT_COMMIT}/mattermost-webapp.tar.gz && mkdir ./dist && tar -xvf ./dist.tar.gz -C ./dist --strip-components=1 || make node_modules test build || exit 1
+                            """
+                        }
+                    }
+                }
+            }
+        }
+
+        stage('Build') {
+            environment {
+                GOPATH = "/go"
+            }
+
+            steps {
+                withDockerContainer(args: '-u root --privileged -v ${WORKSPACE}/src:/go/src/', image: 'mattermost/mattermost-build-server:dec-7-2018') {
+                    ansiColor('xterm') {
+                        sh """
+                            cd /go/src/github.com/mattermost/mattermost-server
+                            mv config/default.json config/config.json || echo ""
+                            make check-style BUILD_NUMBER='${BRANCH_NAME}-${BUILD_NUMBER}' || exit 1
+                            make build BUILD_NUMBER='${BRANCH_NAME}-${BUILD_NUMBER}' || exit 1
+                            make package BUILD_NUMBER='${BRANCH_NAME}-${BUILD_NUMBER}' || exit 1
+                        """
+                    }
+                }
+            }
+        }
+
+        stage('Test') {
+            environment {
+                GOPATH = "/go"
+                TEST_DATABASE_MYSQL_HOSTNAME = "mysql"
+                TEST_DATABASE_MYSQL_PORT = "3306"
+                TEST_DATABASE_POSTGRES_HOSTNAME = "postgres"
+                TEST_DATABASE_POSTGRES_PORT = "5432"
+                TEST_DATABASE_MYSQL_USERNAME = "mmuser"
+                TEST_DATABASE_MYSQL_PASSWORD = "mostest"
+                TEST_DATABASE_MYSQL_NAME = "mattermost_test"
+                TEST_DATABASE_POSTGRES_USERNAME = "mmuser"
+                TEST_DATABASE_POSTGRES_PASSWORD = "mostest"
+                TEST_DATABASE_POSTGRES_NAME = "mattermost_test"
+                CI_INBUCKET_HOST = "inbucket"
+                CI_MINIO_HOST = "minio"
+                CI_INBUCKET_PORT = "10080"
+                CI_MINIO_PORT = "9000"
+                CI_LDAP_HOST = "openldap"
+                IS_CI = true
+            }
+
+            steps {
+                dir('src/github.com/mattermost/mattermost-server/build') {
+                    ansiColor('xterm') {
+                        sh """
+                            COMPOSE_PROJECT_NAME=${DCNAME}-${env.BUILD_NUMBER} docker-compose up -d
+
+                            docker ps
+                        """
+
+                        withDockerContainer(args: "--net ${DCNAME}-${BUILD_NUMBER}_mm-test -v ${WORKSPACE}/src:/go/src/", image: 'postgres:9.4') {
+                            sh """
+                                cd /go/src/github.com/mattermost/mattermost-server/build
+
+                                ./wait-for-it.sh -t 30 openldap-${DCNAME}-${env.BUILD_NUMBER}:389 -- echo "Openldap ready"
```

Comment:
Github burred the feedback, so just reopening that this and below would be nice as part of docker-compose.

Reply:
done @crspeller let me know what do you think